### PR TITLE
Fetch all annotations and labels for pods and namespaces in k8sattributes processor

### PR DESCRIPTION
**Pull request by chaitanyaphalak**
**Description:** 
Added the ability to fetch all annotations and labels from pods and namespace for k8stagger/k8sattributes processor

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/4953

**Testing:** Added relevant Unit tests

**Documentation:** Added relevant docs and config examples

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Review by owais — CHANGES_REQUESTED** [review]

**Review by owais — APPROVED** [review]

File: processor/k8sattributesprocessor/factory.go [hunk 1]
```diff
@@ -233,3 +238,25 @@ func warnDeprecatedMetadataConfig(logger *zap.Logger, cfg config.Processor) {
...
+		}
+	}
+
+	if oCfg.Extract.Annotations != nil {
+		for _, r := range oCfg.Extract.Annotations {
+			if r.Key != "" && r.KeyRegex != "" {
+				return fmt.Errorf("Out of Key or KeyRegex only one option is expected to be configured at a time, currently Key:%s and KeyRegex:%s", r.Key, r.KeyRegex)
+			}
+		}
+	}
```
Comment (dmitryax) [inline processor/k8sattributesprocessor/factory.go:259]: I think this should work:

```
for _, r := range append(oCfg.Extract.Labels, oCfg.Extract.Annotations...) {
	if r.Key != "" && r.KeyRegex != "" {
		return fmt.Errorf("Out of Key or KeyRegex only one option is expected to be configured at a time, currently Key:%s and KeyRegex:%s", r.Key, r.KeyRegex)
	}
}
```

No need for `nil` checks. Please try this out

**Review by dmitryax — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
Some build actions are failing. Please check.

**Comment by chaitanyaphalak** [comment]
@tigrannajaryan fixed some build actions, the rest which are failing will be fixed by this PR https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6091 - looks like they are failing for everyone.

**Comment by tigrannajaryan** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6091 is merged, please rebase to fix the build.

**Comment by chaitanyaphalak** [comment]
@tigrannajaryan its ready!

**Comment by tigrannajaryan** [comment]
Merged. Thanks @chaitanyaphalak 

For future PRs please make sure to populate the commit message (not just one line). The PR description is usually fine as a commit message. I copy/pasted this time.

**Pull request merged**